ra/ra.go

```diff
@@ -2365,9 +2360,7 @@ func (ra *RegistrationAuthorityImpl) NewOrder(ctx context.Context, req *rapb.New
 	// authorizations correspond to
 	nameToExistingAuthz := make(map[string]*corepb.Authorization, len(newOrder.Names))
 	for _, v := range existingAuthz.Authz {
-		// Don't reuse a valid authorization if the reuseValidAuthz flag is
-		// disabled.
-		if v.Authz.Status == string(core.StatusValid) && !ra.reuseValidAuthz {
+		if v.Authz.Status == string(core.StatusValid) {
```

Comment:
Previously this said `if (condition) && !(true)`, which is in other words `if false`. Therefore this entire conditional block needs to be removed, not just remove the reuseValidAuthz half of the boolean statement.

And if the tests don't catch this, we should update the tests to make sure there's something that would have.

Reply:
Good catch, thank you.

Reply:
The check that keeps boulder safe here is https://github.com/letsencrypt/boulder/blob/d3845f25c68061cf9ea1a234230024a86d6ee384/ra/ra_test.go#L2565

In the incorrect RA case of the `if v.Authz.Status == string(core.StatusValid)` conditional being left in, the tests newly created order object would contain the following and fail at `test.AssertEquals(t, numAuthorizations(order), 1)`.
```
id:5790946419980184778 registrationID:1 expires:1425618000000000000 status:"pending" names:"zombo.com" created:1677100167949251813
``